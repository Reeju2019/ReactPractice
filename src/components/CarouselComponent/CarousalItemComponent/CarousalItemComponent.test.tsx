import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import { CarousalItemComponent } from "."

const textAreaHandler = jest.fn()

/**
 * Unit testing for CarousalItem Component
 */
describe("Carousel Item Component Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <CarousalItemComponent
        defaultInterval={5000}
        textAreaHandler={textAreaHandler}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <CarousalItemComponent
        defaultInterval={5000}
        textAreaHandler={textAreaHandler}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
