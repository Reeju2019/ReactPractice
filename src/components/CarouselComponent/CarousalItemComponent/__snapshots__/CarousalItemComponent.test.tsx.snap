// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item Component Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Row
      className="def-img"
    />
    <CarouselCaption>
      <FormLabel
        column={false}
        visuallyHidden={false}
      >
        Default HEADER
      </FormLabel>
      <FormLabel
        column={false}
        visuallyHidden={false}
      >
        Default Caption
      </FormLabel>
      <React.Fragment />
    </CarouselCaption>
  </React.Fragment>,
]
`;
